ecllibrary/std/Date.ecl

```diff
@@ -764,6 +764,32 @@ EXPORT STRING SecondsToString(Seconds_t seconds, VARSTRING format = '%Y-%m-%dT%H
     TimeLib.SecondsToString(seconds, format);
 
 
+/**
+ * Converts a Timestamp_t value into a human-readable string using a format template.
+ *
+ * @param timestamp     The microseconds since epoch.
+ * @param format        The format template to use for the conversion; see
+ *                      strftime() for appropriate format specifiers.  Two
+ *                      additional format specifiers are available to show
+ *                      fractional seconds:
+ *                          %@ - fraction of seconds in microseconds (6 digits)
+ *                          %# - fraction of seconds in milliseconds (3 digits)
+ *                      The maximum length of the resulting string is 255
+ *                      characters.  This parameter is optional and defaults to
+ *                      '%Y-%m-%dT%H:%M:%S.%@' which is YYYY-MM-DDTHH:MM:SS.ssssss.
+ * @return              The converted timestamp as a string.
+ */
+
+EXPORT STRING TimestampToString(Timestamp_t timestamp, VARSTRING format = '%Y-%m-%dT%H:%M:%S.%@') := FUNCTION
+    f := (UNSIGNED4)(timestamp % 1000000);
+    s1 := TimeLib.SecondsToString(timestamp DIV 1000000, format);
+    s2 := REGEXREPLACE('%@', s1, (STRING)f);
```

Comment:
This is not correct - if the milli seconds are 1 then it will output as .1 rather than .001.  You probably want to use INTFORMAT.

The PR should gain some more test cases - 0, 1, 999, 1000 and 999999 for the microsecond portion at least.